src/main/java/com/fasterxml/jackson/databind/deser/impl/PropertyValueBuffer.java

```diff
@@ -135,6 +137,12 @@ public PropertyValueBuffer(JsonParser p, DeserializationContext ctxt, int paramC
      */
     public final boolean hasParameter(SettableBeanProperty prop)
     {
+        // 28-Sep-2024 : [databind#4508] Support any-setter flowing through creator
+        if (_anyParamSetter != null) {
+            if (prop.getCreatorIndex() == _anyParamSetter.getParameterIndex()) {
+                return true;
+            }
+        }
```

Comment:
I am not sure if this is the most optimal way of checking, any idea?

Reply:
Let me rewrite it slightly, considering that expectation is that match (true) is more likely than non-match.

Reply:
Rewrite looks cleaner, thanks!